src/main/java/com/minecolonies/entity/ai/citizen/builder/EntityAIStructureBuilder.java

```diff
@@ -688,64 +659,57 @@ private boolean placeBlock(BlockPos pos, Block block, IBlockState metadata)
+            EnumFacing facing = blockState.getValue(BlockBed.FACING);
 
             //Set other part of the bed, to the opposite PartType
-            if (metadata.getValue(BlockBed.PART) == BlockBed.EnumPartType.FOOT)
+            if (blockState.getValue(BlockBed.PART) == BlockBed.EnumPartType.FOOT)
             {
                 //pos.offset(facing) will get the other part of the bed
-                world.setBlockState(pos.offset(facing), metadata.withProperty(BlockBed.PART, BlockBed.EnumPartType.HEAD), 0x03);
+                world.setBlockState(pos.offset(facing), blockState.withProperty(BlockBed.PART, BlockBed.EnumPartType.HEAD), 0x03);
             }
             else
             {
-                world.setBlockState(pos.offset(facing), metadata.withProperty(BlockBed.PART, BlockBed.EnumPartType.FOOT), 0x03);
+                world.setBlockState(pos.offset(facing), blockState.withProperty(BlockBed.PART, BlockBed.EnumPartType.FOOT), 0x03);
             }
         }
         else if (block instanceof BlockDoublePlant)
         {
-            world.setBlockState(pos, metadata.withProperty(BlockDoublePlant.HALF, BlockDoublePlant.EnumBlockHalf.LOWER), 0x03);
-            world.setBlockState(pos.up(), metadata.withProperty(BlockDoublePlant.HALF, BlockDoublePlant.EnumBlockHalf.UPPER), 0x03);
+            world.setBlockState(pos, blockState.withProperty(BlockDoublePlant.HALF, BlockDoublePlant.EnumBlockHalf.LOWER), 0x03);
+            world.setBlockState(pos.up(), blockState.withProperty(BlockDoublePlant.HALF, BlockDoublePlant.EnumBlockHalf.UPPER), 0x03);
         }
         else if (block instanceof BlockEndPortal || block instanceof BlockMobSpawner || block instanceof BlockDragonEgg || block instanceof BlockPortal)
         {
             return true;
         }
         else
         {
-            if (!world.setBlockState(pos, metadata, 0x03))
+            if (!world.setBlockState(pos, blockState, 0x03))
             {
                 return false;
             }
-            if (world.getBlockState(pos).getBlock() == block && world.getBlockState(pos) != metadata)
+            if (world.getBlockState(pos).getBlock() == block && world.getBlockState(pos) != blockState)
             {
-                world.setBlockState(pos, metadata, 0x03);
+                world.setBlockState(pos, blockState, 0x03);
             }
         }
 
-        ItemStack stack = new ItemStack(Item.getItemFromBlock(block), 1, block.damageDropped(metadata));
-
-        if (stack.getItem() == null)
-        {
-            stack = new ItemStack(block);
-        }
-
-        if (stack.getItem() == null)
+        ItemStack stack = BlockUtils.getItemStackFromBlockState(blockState);
+        if (stack == null)
         {
-            stack = new ItemStack(BlockUtils.getItemFromBlock(block));
+            Log.logger.warn("Block causes NPE: " + blockState.getBlock());
         }
 
         int slot = worker.findFirstSlotInInventoryWith(stack.getItem());
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) NullPointerException might be thrown as 'stack' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS2259)